### PR TITLE
ARROW-2759: [Plasma] Export plasma notification socket

### PR DESCRIPTION
**Pull request by suquark**
https://issues.apache.org/jira/browse/ARROW-2759

### CODE REVIEW

**Comment by suquark** [comment]
Ready to merge. @pcmoritz Can you review this PR?

File: python/pyarrow/_plasma.pyx [hunk 1]
```diff
@@ -729,6 +735,44 @@ cdef class PlasmaClient:
         with nogil:
             check_status(self.client.get().Subscribe(&self.notification_fd))
 
+    def get_notification_socket(self):
+        """
+        Get the notification socket.
+        """
+        if PY_MAJOR_VERSION >= 3:
```
Comment (pcmoritz) [inline python/pyarrow/_plasma.pyx]: This code should go into compat.py

File: python/pyarrow/tests/test_plasma.py [hunk 2]
```diff
@@ -742,6 +743,34 @@ def test_subscribe(self):
...
+                self.plasma_client.create(
+                    object_ids[j], data_sizes[j],
+                    metadata=bytearray(np.random.bytes(metadata_sizes[j])))
+                self.plasma_client.seal(object_ids[j])
+
+            # Check that we received notifications for all of the objects.
+            for j in range(i):
+                # Assume the plasma store will not be full,
+                # so we always get the data size instead of -1.
+                msg_len = struct.unpack('L', rsock.recv(8))[0]
```
Comment (pcmoritz) [inline python/pyarrow/tests/test_plasma.py]: This looks uggly, can we call GetNotification here instead?

**Comment by suquark** [comment]
Ready to merge. The test fail is not because of this PR.

**Review by pcmoritz — APPROVED** [review]
+1 The test failure is unrelated

**Pull request closed without merging**